Bloodpacks can no longer be ordered from cargo, and can no longer be generated from NPCs or disconnected player

Metekillot opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

The title explains the entirety of what this pull request does.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

From my observing, NPCs are treated as mindless bloodpack factories rather than players making an attempt to treat them as if they're human being who are part of the world. This is an immense negative to immersion. Additionally, these infinite bloodpack factories totally destroy the dynamic of vampires needing to feed on the blood of the living to survive; instead, everyone is just carrying around mana potions instead of actually hunting humans for blood.

This change will make it so that bloodpacks can only be (for now) gained from players who consent or are otherwise coerced into donating blood at the machines. Right now the machines don't actually do anything to your real blood volume; it simply lowers your bloodpool (which, incidentally for humans, is actually totally separate from your body's blood volume and seemingly doesn't interact with it at all...). That will change in the future but for now, to get your mana potions, you need to get them from a player, and I don't mean over the counter.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and its effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Bisar
balance: Bloodpacks can now only be created from players with active minds; they cannot be gained from NPCs, disconnected players, or ordered from cargo.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

UltraBigRat commented:
Maybe keep it allowed to extract blood from npcs, or make blood packs more expensive? Theres too few humans to make a good blood supply and there are characters out there who avoid nonconsensual feeding. (Which all npc feeding would be)

Potatocrip commented:
I _agree_ that just turning NPCs into bloodpack machines is LRP, but this is a _**HUGE**_ impact on characters who only drink from consenting parties, whether by choice or by clan curse. Not enough players go to the clinic to donate blood because it's not always clear that doing so is an option, and even when it is, not many players want to anyways. 
Completely destroying the bloodpack market like this means consentualists are left with no other option but to somehow have rapport with enough ghouls in a given round to give 3-4 BP each until their pool is filled, with enough to spare to be able to ask for more later when needed. And there are _not_ enough ghoul players to be able to do that with.
That's not even mentioning the fact that reviving Torpor'd kindred without bloodbonding them can ONLY be done through feeding bloodpacks.
This is a miserable TG-like fix to something that's barely even an issue to begin with.

K4rlox commented:
this feels like a PR on TG that would get 172 thumbsdowns and 21 thumbsups and still get merged

K4rlox commented:
i would like to also say this person closed my pr doing the exact opposite of this, and as soon as they closed it, opened this PR, doing the exact opposite of my pr
![image](https://github.com/user-attachments/assets/231d4990-7e68-4b16-b27e-e574a98b2077)
![image](https://github.com/user-attachments/assets/8b048696-d1e4-4351-a080-3d295981ffca)

Metekillot commented:
We actually have a policy against self-merging and I understand it may feel retaliatory for me to create this so quickly after closing your vitae PR. That's not my intent and I honestly should have considered the timing when I threw it together. I apologize because even if it wasn't deliberate, it still sends a very rude message and that didn't even occur to me.

> I _agree_ that just turning NPCs into bloodpack machines is LRP, but this is a _**HUGE**_ impact on characters who only drink from consenting parties, whether by choice or by clan curse. Not enough players go to the clinic to donate blood because it's not always clear that doing so is an option, and even when it is, not many players want to anyways. Completely destroying the bloodpack market like this means consentualists are left with no other option but to somehow have rapport with enough ghouls in a given round to give 3-4 BP each until their pool is filled, with enough to spare to be able to ask for more later when needed. And there are _not_ enough ghoul players to be able to do that with. That's not even mentioning the fact that reviving Torpor'd kindred without bloodbonding them can ONLY be done through feeding bloodpacks. This is a miserable TG-like fix to something that's barely even an issue to begin with.

Vampires can use the blood machines. My intent is for gaining blood to require the gameplay loop, instead of our current implementation of infinite cargo mana potions. Blood management is meant to be an core part of the tabletop experience, but not just the tabletop experience, the theme itself. Minimizing the acquirement of blood to buying it from the Dealer totally annihilates that. If you absolutely insist on playing a vampire who only drinks from the willing and our low human numbers make that an issue, that's an issue with the incentives for people to play human.

As a matter of fact, this change actually DOES incentivize people playing human. Vampires have no reason to give a shit about human players because they offer nothing to them; NPCs are easier to feed on and bloodbags are effortless to acquire. Don't get me wrong, playing a human will still be boring if not miserable in comparison to the dozens of powers that vampires get (with few to none of the downsides because we currently don't have daytime gameplay, fire is barely a bane, true faith and hunters aren't implemented), but it's a good start.

It's going to take an act of god for me to budge my view on extracting bloodpacks from NPCs. I'd also be fine with increasing the amount of bloodpacks you get from the machine if someone donates, but tacitly condoning totally ignoring what should immersively be treated as another person so you can keep your mana up is really out of the question.

> Removing them from cargo is absolutely TERRIBLE balance-wise. I suggest you instead make them limited supply and increase the price. Maybe three or so orders of elite bags a round and five or so standard bags a round. People should still be able to buy blood.

I don't agree, regarding balance. You're not supposed to be able to access a full bloodpool at all times. There's no reason for you to actually carefully consider the use of your powers with bloodpacks being readily available in the way they are now. Maybe I'll consider being able to coordinate a raid on a shipment of bloodbags with the assistance of doctor players down the line, or I may be persuaded to increase the bloodbags gained from the machine ( since one every two minutes very truly doesn't seem exactly sustainable in a server of ~90-100 ), but having them be something that can be ordered at will and with ease isn't something I'm interested in doing. You need to engage in roleplay scenes for bloodbags and that's something I'm not budging on.



> Maybe keep it allowed to extract blood from npcs, or make blood packs more expensive? Theres too few humans to make a good blood supply and there are characters out there who avoid nonconsensual feeding. (Which all npc feeding would be)

Regarding the nonconsensual feeding bit: you can feed on vampires, too, you know. And there being so few humans is an issue with the experience of human players and we can't address every issue simultaneously. Players aren't treating access to blood with the weight that should be appropriate and the primary driver of that is easy access to bloodbags through NPC factory farms and the cargo shipments.

So, let me summarize my points:

## Blood management is a core part of the experience of playing a vampire. There is none of that right now; the only concern is finding ~2 minutes to guzzle your mana potions after fights.
## NPC factory farms are a devastation to our steady direction toward the world being responsive and immersive. I may consider allowing NPCs to be extracted from, down the line, once we more robustly implement social skills. But as of our infrastructure right now, it's not tenable.
## Vampires have no mechanical reasons whatsoever to have any consideration for human players. This isn't the primary motivator of this change, but it will exert an influence on the human<->vampire interactions for both sides to know the human has something the vampire might actually want, now. The current dynamic? Why give a shit about humans, I have 14 bloodpacks!
## This is something of a reiteration of my first point but: spending blood is not something you should be doing lightly. This is intended to have a cooling effect on people going super saiyan celerity potence 5 at every opportunity as a result of a conflict. Sometimes, both sides are better off baring fangs and puffing chests before fucking off instead of duking it out because they know blood will be easily restored.

Lucia's rework of Disciplines is pending completion, which will help me speed up my work on how splats are applied instead of their current (terrible) implementation. Though there's a freeze on splat-specific powers, there is no freeze on the development of mechanics for people to interact with conflict that isn't physical violence. It's part of our development roadmap to have social skills be able to be used both to help and harm other players, and we're not referring to the currently overpowered Dominate and Presence skills.

cheapcents commented:
Hey there Metek,

Making blood bags harder to obtain is definitely the way to go, but we first should implement more ways to engage in non-lethal combat for the kidnapping of other players before we take away the only practical way of producing blood we have.

My suggestion: Make it so extracting blood from NPCs instead gives you "wastrel blood": blood of a much lower quality than that of a player's

Ping me on discord if you agree. I can make you some sprites for all the different tiers of blood bags

Metekillot commented:
> Hey there Metek,
> 
> Making blood bags harder to obtain is definitely the way to go, but we first should implement more ways to engage in non-lethal combat for the kidnapping of other players before we take away the only practical way of producing blood we have.
> 
> My suggestion: Make it so extracting blood from NPCs instead gives you "wastrel blood": blood of a much lower quality than that of a player's
> 
> Ping me on discord if you agree. I can make you some sprites for all the different tiers of blood bags

I'm troubled that overt, violent conflict is the primary means with which you think people are going to have to acquire blood. I think I robustly addressed concerns about it in my reply above, if you want to brainstorm on ways for players to acquire blood that is engaging roleplay-wise.

KittyNoodle commented:
> I _agree_ that just turning NPCs into bloodpack machines is LRP, but this is a _**HUGE**_ impact on characters who only drink from consenting parties, whether by choice or by clan curse. Not enough players go to the clinic to donate blood because it's not always clear that doing so is an option, and even when it is, not many players want to anyways. Completely destroying the bloodpack market like this means consentualists are left with no other option but to somehow have rapport with enough ghouls in a given round to give 3-4 BP each until their pool is filled, with enough to spare to be able to ask for more later when needed. And there are _not_ enough ghoul players to be able to do that with. That's not even mentioning the fact that reviving Torpor'd kindred without bloodbonding them can ONLY be done through feeding bloodpacks. This is a miserable TG-like fix to something that's barely even an issue to begin with.

Chosing to be a consensualist should actually be a hard mode when it comes to feeding. You should be encouraged to have to build up rapport with mortals in order to feed, because that's half the point of being a consensualists.

Potatocrip commented:
> Chosing to be a consensualist should actually be a hard mode when it comes to feeding. You should be encouraged to have to build up rapport with mortals in order to feed, because that's half the point of being a consensualist.
That makes sense, yeah-- But I think there should be more direct changes to incentivize playing a mortal/ghoul, _before_ such a large change that hinges on there being more mortal/ghoul players is made. I think there _needs_ to be more content in-game that facilitates _**human to human**_ interaction first and foremost before there can be healthier _human to kindred_ interaction. 
Like, for as much as people clown on bar RP, there's a reason it happens as much as it does-- there's a _reason_ the Anarch bar is as populated as it always is: It Works. People love having a room where they can sit down and spend money on functionally useless stuff to facilitate being able to talk and interact with other players. But right now, the only places where that can happen are in kindred-owned establishments: the Anarch bar run by the Anarchs, the strip club run by the Toreador, the coffee shop run by the Banu Haqim, the Tower bar run by the Camarilla, the Cabaret run by the Giovanni (I think? The Gio keys work on those doors, no?), etc. 

This is a huge detachment from the content of this particular PR, but it's all to say that I think there should be more mechanically-supported roles for humans to play that are focused on _human-to-human_ interaction for kindred to be able to slip into, to give players more incentive to _play humans_--. Make roles for employees of different venues that give them the keys to those buildings. I find it hard to believe that the kindred would own _EVERY VENUE IN SAN FRANCISCO._
Once those incentives are made, _then_ reinforce the interaction by forcing kindred players to rely on human ones by making bloodpacks less accessible. I think this is an okay change with some very solid reasoning behind it that's just timed really poorly, and shouldn't be implemented right now.

DatFrostys commented:
I don't have any strong opinions about the changes though I would like to point out the fact this will make blood-pack related crafts for clans such as Tzimisce quite annoying.

Metekillot commented:
Upon discussion with the other maintainers, we won't be proceeding with this.
